homeassistant/components/shelly/cover.py

```diff
@@ -0,0 +1,98 @@
+"""Cover for Shelly."""
+from aioshelly import Block
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    SUPPORT_CLOSE,
+    SUPPORT_OPEN,
+    SUPPORT_SET_POSITION,
+    SUPPORT_STOP,
+    CoverEntity,
+)
+from homeassistant.core import callback
+
+from . import ShellyBlockEntity, ShellyDeviceWrapper
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up cover for device."""
+    wrapper = hass.data[DOMAIN][config_entry.entry_id]
+    blocks = [block for block in wrapper.device.blocks if block.type == "roller"]
+
+    if not blocks:
+        return
+
+    async_add_entities(ShellyCover(wrapper, block) for block in blocks)
+
+
+class ShellyCover(ShellyBlockEntity, CoverEntity):
+    """Switch that controls a cover block on Shelly devices."""
+
+    def __init__(self, wrapper: ShellyDeviceWrapper, block: Block) -> None:
+        """Initialize light."""
+        super().__init__(wrapper, block)
+        self.control_result = None
+        self._supported_features = SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP
+        if self.wrapper.device.settings["rollers"][0]["positioning"]:
+            self._supported_features |= SUPPORT_SET_POSITION
+
+    @property
+    def is_closed(self):
+        """If cover is closed."""
+        if self.control_result:
+            return self.control_result["rollerPos"] == 0
+
+        return self.block.rollerPos == 0
+
+    @property
+    def current_cover_position(self):
+        """Position of the cover."""
+        if self.control_result:
+            return self.control_result["rollerPos"]
```

Comment:
Should be current_pos, see https://shelly-api-docs.shelly.cloud/#shelly2-5-roller-index
```suggestion
            return self.control_result["current_pos"]
```